make.mk

```diff
@@ -242,3 +242,37 @@ endif
 ifdef CLOCK_FACE_24H_ONLY
 CFLAGS += -DCLOCK_FACE_24H_ONLY
 endif
+
+# DATE = X
+#  YEAR = Sets the year and timezone to the PC's
+#  DAY = Sets the default time down to the day (year, month, day, timezone)
+#  MIN = Sets the default time down to the minute (year, month, day, timezone, hour, minute)
+ifdef DATE
+TIMEZONE := $(shell date +%z | awk '{print substr($$0, 1, 3) * 60 + substr($$0, 4, 2)}')
+CURRENT_YEAR := $(shell echo $$(($(shell date +"%Y") - 2020)))
+CURRENT_MONTH := $(shell date +"%-m")
+CURRENT_DAY := $(shell date +"%-d")
+CURRENT_HOUR := $(shell date +"%-H")
+CURRENT_MINUTE := $(shell date +"%-M")
+ifeq ($(DATE), YEAR)
+CFLAGS += -DMAKEFILE_TIMEZONE=$(TIMEZONE)
+CFLAGS += -DMAKEFILE_CURR_YEAR=$(CURRENT_YEAR)
+$(info Default year and timezone are set to $(shell date +"%Y") $(shell date +%Z))
+else ifeq ($(DATE), DAY)
+CFLAGS += -DMAKEFILE_TIMEZONE=$(TIMEZONE)
+CFLAGS += -DMAKEFILE_CURR_YEAR=$(CURRENT_YEAR)
+CFLAGS += -DMAKEFILE_CURR_MONTH=$(CURRENT_MONTH)
+CFLAGS += -DMAKEFILE_CURR_DAY=$(CURRENT_DAY)
+$(info Default date set to $(shell date +"%b") $(CURRENT_DAY) $(shell date +"%Y") $(shell date +%Z))
+else ifeq ($(DATE), MIN)
+CFLAGS += -DMAKEFILE_TIMEZONE=$(TIMEZONE)
+CFLAGS += -DMAKEFILE_CURR_YEAR=$(CURRENT_YEAR)
+CFLAGS += -DMAKEFILE_CURR_MONTH=$(CURRENT_MONTH)
+CFLAGS += -DMAKEFILE_CURR_DAY=$(CURRENT_DAY)
+CFLAGS += -DMAKEFILE_CURR_HOUR=$(CURRENT_HOUR)
+CFLAGS += -DMAKEFILE_CURR_MINUTE=$(CURRENT_MINUTE)
```

Comment:
I suggest not mentioning `MAKEFILE` in these variables. Users might add them to a header file and include it instead. The makefile therefore wouldn't be the only source of these values. See also PR #295 for example.

I suggest `INITIAL_TIMEZONE`, `INITIAL_YEAR`, and so on.